libecole/src/observation/strongbranchingscores.cpp

```diff
@@ -0,0 +1,102 @@
+		"branching/vanillafullstrong/integralcands");
+	bool scoreall = model.get_param_explicit<scip::ParamType::Bool>(
+		"branching/vanillafullstrong/scoreall");
+	bool collectscores = model.get_param_explicit<scip::ParamType::Bool>(
+		"branching/vanillafullstrong/collectscores");
+	bool donotbranch = model.get_param_explicit<scip::ParamType::Bool>(
+		"branching/vanillafullstrong/donotbranch");
+	bool idempotent = model.get_param_explicit<scip::ParamType::Bool>(
+		"branching/vanillafullstrong/idempotent");
+
+	/* set parameters for vanilla full strong branching  */
+	SCIP* scip = model.get_scip_ptr();
+
+	scip::call(SCIPsetBoolParam, scip, "branching/vanillafullstrong/integralcands", true);
+	scip::call(SCIPsetBoolParam, scip, "branching/vanillafullstrong/scoreall", true);
+	scip::call(SCIPsetBoolParam, scip, "branching/vanillafullstrong/collectscores", true);
+	scip::call(SCIPsetBoolParam, scip, "branching/vanillafullstrong/donotbranch", true);
+	scip::call(SCIPsetBoolParam, scip, "branching/vanillafullstrong/idempotent", true);
+
+	/* execute vanilla full strong branching */
+	SCIP_BRANCHRULE* branchrule = SCIPfindBranchrule(scip, "vanillafullstrong");
+	SCIP_Bool allowaddcons = TRUE;
+	SCIP_RESULT result = SCIP_FEASIBLE;
+
+	branchrule->branchexeclp(scip, branchrule, allowaddcons, &result);
+
+	/* get vanilla full strong branching scores */
+	SCIP_VAR** cands;
+	SCIP_Real* candscores;
+	int ncands;
+	int npriocands;
+	int bestcand;
+
+	SCIPgetVanillafullstrongData(
+		scip, &cands, &candscores, &ncands, &npriocands, &bestcand);
+
+	assert(ncands >= 0);
+	assert(npriocands >= 0);
+	assert(ncands == npriocands);
+	assert(bestcand > -1);
+
+	/* restore model parameters */
+	scip::call(
+		SCIPsetBoolParam, scip, "branching/vanillafullstrong/integralcands", integralcands);
+	scip::call(SCIPsetBoolParam, scip, "branching/vanillafullstrong/scoreall", scoreall);
+	scip::call(
+		SCIPsetBoolParam, scip, "branching/vanillafullstrong/collectscores", collectscores);
+	scip::call(
+		SCIPsetBoolParam, scip, "branching/vanillafullstrong/donotbranch", donotbranch);
+	scip::call(
+		SCIPsetBoolParam, scip, "branching/vanillafullstrong/idempotent", idempotent);
+
+	/* Store strong branching scores in tensor */
+	int num_lp_columns = SCIPgetNLPCols(scip);
+	auto strong_branching_scores =
+		xt::xarray<double>::from_shape({static_cast<unsigned long>(num_lp_columns)});
+	strong_branching_scores.fill(std::nan(""));
+
+	SCIP_COL* col;
+	int lp_index;
+	for (int i = 0; i < ncands; i++) {
+		col = SCIPvarGetCol(cands[i]);
+		lp_index = SCIPcolGetLPPos(col);
+		strong_branching_scores(lp_index) = static_cast<double>(candscores[i]);
+	}
+
+	return strong_branching_scores;
+}
+
+auto StrongBranchingScores::obtain_observation(environment::State const& state)
+	-> nonstd::optional<StrongBranchingScoresObs> {
+
+	if (state.model.get_stage() == SCIP_STAGE_SOLVING) {
+		return StrongBranchingScoresObs{extract_strong_branching_scores(state.model)};
+	} else {
+		return {};
```

Comment:
I would raise an Exception here, rather than silently ignoring the problem.
I think you can just call extract_strong_branching_scores, and let SCIP raise an Exception by itself. That should be tested though.

Reply:
I agree it may be possible to just call it without the condition, however in NodeBipartite it is implemented in the same way so maybe Antoine can comment on a reason for that.

Reply:
I tested it and we will get an exception when SCIP is finished solving (i.e. `state.model.get_stage() == SCIP_STAGE_SOLVED`).  This is probably because `obtain_observation()` is called in the last step when we would return 'done' as 'True'.

Reply:
We'll tackle that in #59 so you can leave as is for this PR.